### ISSUE TITLE
Migration of Eclipse platform TLP to Github

### ISSUE THREAD

**Issue by sravanlakkimsetti** [eclipse-platform/eclipse.platform.releng.aggregator#24]
Here is the list of repositories that needs to be migrated to github. Completed ones will be checked

- [x] #99
- [x] #121
- [x] #25
- [x] [eclipse.platform.releng.aggregator.git](https://git.eclipse.org/c/platform/eclipse.platform.releng.aggregator.git/)
- [x] [eclipse.platform.releng.buildtools.git](https://git.eclipse.org/c/platform/eclipse.platform.releng.buildtools.git/)
- [x] [eclipse.platform.releng.git](https://git.eclipse.org/c/platform/eclipse.platform.releng.git/)
- [x] #146 
- [x] #149
- [x] #100
- [x] #125
- [x] #123
- [x] #142
- [x] #144
- [x] #103 
- [x] #102 
- [x] #162
- [x] #161 
- [x] [eclipse.platform.text.git](https://git.eclipse.org/c/platform/eclipse.platform.text.git/)
- [x] #163
- [x] #176
- [x] #177
- [x] #200
- [x] #201
- [x] #210

### Migration Process 
(Courtesy @mickaelistria)

**Prerequisites**
The repo must have very few Gerrit patches pending. Migrating existing contributions from Gerrit to GitHub will be annoying for submitters and reviewers, so we should avoid this as much as possible

**Migration steps**

1. Request move of repo to GitHub eg https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/891
2. Once GitHub repo is available, go to its settings and customize it. For the moment
    - Disable “Projects”
    - Disable “Wiki”
    - Disable “Merge commits”
3. Once GitHub repo is available, submit PR to aggregator to update submodule URL, eg https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/19
4. Upon successful completion of the checks, merge the PR. The GitHub repo is now the reference one that’s used to build Eclipse
5. Update CI job to reference GitHub repo, from the Configuration page, replace Gerrit with GitHub. The result looks like this: https://ci.eclipse.org/platform/job/eclipse.platform.releng.aggregator/jobConfigHistory/showDiffFiles?timestamp1=2022-02-09_10-40-07&timestamp2=2022-02-10_09-55-29
6. Update root pom.xml to reference GitHub repo in `tycho.scmUrl` property eg https://github.com/eclipse-platform/eclipse.platform.releng/pull/3
7. Update Jenkinsfile for the repo to not include anything related to Gerrit, eg https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/commit/6de2cdcecbcf39cd4a810aec801b2b079493fbb3
8. ~~For the Gerrit repo, create a commit that removes all content and add a file describes the move to GitHub, eg https://git.eclipse.org/r/c/platform/eclipse.platform.releng.aggregator/+/190436 , see https://git.eclipse.org/c/platform/eclipse.platform.releng.aggregator.git/tree/ for result. Message in the remaining file should include steps to perform migration, eg~~
```
This repository is now moved to GitHub: https://github.com/eclipse-platform/eclipse.platform.releng.aggregator
If you see this message, perform GitHub migration by (assuming the legacy Gerrit repo is called `origin`)
$ git reset --hard HEAD^
$ git remote set upstream git@github.com:eclipse-platform/eclipse.platform.releng.git
$ git pull upstream master
```
~~(dot files can/should be removed as well)~~ Do not modify Gerrit repo to not break users (see https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/24#issuecomment-1042949957 )
9 Update https://projects.eclipse.org/projects/eclipse.platform/developer to point to GitHub repo instead of Gerrit one. 
10. Send a notification to the project mailing-list + eclipse-dev mailing-list about moved repository.
```
This Git repository is now moved to GitHub: https://github.com/eclipse-platform/eclipse.platform.releng.aggregator . Please immediately set you `upstream` repo url to use GitHub instead of Gerrit
$ git remote set-url upstream git@github.com:eclipse-platform/eclipse.platform.releng.git

Also update your development files (Oomph profiles, *.psf files...) to adjust your workflow.
```
11. Close all remaining open Gerrit issues with a message explaining the move to GitHub and how to transition, eg
```
This repository is now moved to GitHub: https://github.com/eclipse-platform/eclipse.platform.releng.aggregator . Please immediately set you `upstream` repo url to use GitHub instead of Gerrit
$ git remote set-url upstream git@github.com:eclipse-platform/eclipse.platform.releng.git

If you're not planning to continue working on this patch; please mark this current Gerrit review as "Abandoned".
If you're willing to transfer this issue to GitHub Pull Requests:
1. On GitHub Web Interface, fork the repository, and retrieve the URL for the fork eg `git@github.com:my-github-userid/eclipse.platform.releng.git`
2. $ git remote add origin git@github.com:my-github-userid/eclipse.platform.releng.git
3. Checkout this current patch (See the "Download" link on this Gerrit review page)
4. $ git push origin HEAD:refs/heads/a-meaningful-branch-name
5. As output of the push, Git shows a link to Create a Pull Request, follow this link and create the pull request. If some useful discussion took place on Gerrit, you may want to include a link to the Gerrit review on the GitHub pull request message, for reference.
```

**Comment by vogella** [comment]
@sravanlakkimsetti what is the process of moving a repo?

**Comment by sravanlakkimsetti** [comment]
> @sravanlakkimsetti what is the process of moving a repo?

Updated the process in the issue description.

**Comment by mickaelistria** [comment]
Will we be able to continue this migration during RCs, or are we now to wait for 4.23 completion?

**Comment by sravanlakkimsetti** [comment]
Lets wait for 4.23 completion and target all the platform repos by 4.24 M1

**Comment by jukzi** [comment]
OOmphing a JDT+Platform+PDE workspace fails:  'org.eclipse.equinox.p2.iu; org.eclipse.platform.releng.tychoeclipsebuilder.plain.project 0.0.0' but it could not be found

is that related?


`Executing startup tasks
OpenJDK Runtime Environment 17.0.2+8
Product org.eclipse.products.epp.package.committers.latest
Workspace C:\Users\jkubitz\20220216\latestRelease platformWs
Project org.eclipse.jdt.features.master
Project org.eclipse.jdt.core.master
Project org.eclipse.jdt.debug.master
Project org.eclipse.jdt.ui.master
Project org.eclipse.jdt.core.tests.master
Project org.eclipse.jdt.core.testbinaries.master
Project org.eclipse.jdt.debug.tests.master
Project org.eclipse.jdt.ui.tests.master
Project org.eclipse.jdt.ui.examples.master
Project org.eclipse.pde.ui.master
Project org.eclipse.pde.build.master
Project org.eclipse.platform.platform.master
Project org.eclipse.platform.debug.master
Project org.eclipse.platform.documentation.master
Project org.eclipse.platform.news.master
Project org.eclipse.platform.releng.master
Project org.eclipse.platform.releng.aggregator.master
Project org.eclipse.platform.resources.master
Project org.eclipse.platform.runtime.master
Project org.eclipse.platform.swt.master
Project org.eclipse.platform.team.master
Project org.eclipse.platform.text.master
Project org.eclipse.platform.ua.master
Project org.eclipse.platform.ui.master
Project org.eclipse.platform.ui.tools.master
Project org.eclipse.platform.dependencies.ecf.master
Project org.eclipse.platform.dependencies.emf.master
Bundle org.eclipse.oomph.setup 1.22.0.v20211015-0620, build=5238, branch=a12cce69e7a50896d22dde5e2e678b4f7b46aa08
Bundle org.eclipse.oomph.setup.core 1.23.0.v20220106-0522, build=5330, branch=b665be4c7396e8a562d18975557797758f1a071f
Bundle org.eclipse.oomph.setup.git 1.20.0.v20211007-0506, build=5211, branch=60f955f75c99caaef0d09892bb6ffee4472287bd
Bundle org.eclipse.oomph.setup.jdt 1.17.0.v20210221-1019, build=5169, branch=4f2d83d895798b31dc987dc9c07da7c94b536f1d
Bundle org.eclipse.oomph.setup.p2 1.18.0.v20220106-0522, build=5330, branch=b665be4c7396e8a562d18975557797758f1a071f
Bundle org.eclipse.oomph.setup.pde 1.14.0.v20220127-0606, build=5352, branch=2a1fed60fb268b2328c2eda3be3cb049bd140a2e
Bundle org.eclipse.oomph.setup.targlets 1.13.0.v20200624-1156, build=5150, branch=947f82e0e8d094222430b7a2cc87a288182f7ca3
Bundle org.eclipse.oomph.setup.workingsets 1.14.0.v20211015-0620, build=5238, branch=a12cce69e7a50896d22dde5e2e678b4f7b46aa08
Performing Targlets Modular Target (ECF Dependency + EMF Dependency + Eclipse Platform + JDT Core + JDT Core Test Binaries + JDT Core Tests + JDT Debug + JDT Debug Tests + JDT Features + JDT Platform + JDT UI + JDT UI Examples + JDT UI Tests + PDE + PDE Build + PDE UI + Platform + Platform Base, Update, and Ant + Platform Debug + Platform Documentation + Platform News + Platform Releng + Platform Releng Aggregator + Platform Resources + Platform Runtime + Platform Team + Platform Text + Platform UA + Platform UI + Platform UI Tools + SWT + SWT Binaries), activate
Repository https://download.eclipse.org/eclipse/updates/4.23-I-builds
Repository https://download.eclipse.org/cbi/updates/license
Repository https://download.eclipse.org/reddeer/releases/3.5.0
Repository https://download.eclipse.org/reddeer/releases/latest
Repository https://download.eclipse.org/cbi/updates/license
Repository https://download.eclipse.org/tools/orbit/downloads/latest-R
Repository https://download.eclipse.org/egit/updates
Repository https://download.eclipse.org/modeling/emf/emf/builds/milestone/latest
Repository https://download.eclipse.org/tools/gef/updates/milestones
Repository https://download.eclipse.org/nebula/snapshot
Repository https://download.eclipse.org/modeling/emf/emf/builds/milestone/latest
Repository https://download.eclipse.org/tools/gef/updates/milestones
Offline = false
Mirrors = true
Include All Platforms = false
Include All Requirements = true
Include Binary Equivalents = true
Include Sources = true
Adding repository https://download.eclipse.org/eclipse/updates/4.23-I-builds
Adding repository https://download.eclipse.org/reddeer/releases/3.5.0
Adding repository https://download.eclipse.org/reddeer/releases/latest
Adding repository https://download.eclipse.org/egit/updates
Adding repository https://download.eclipse.org/cbi/updates/license
Adding repository https://download.eclipse.org/modeling/emf/emf/builds/milestone/latest
Adding repository https://download.eclipse.org/tools/orbit/downloads/latest-R
Adding repository https://download.eclipse.org/tools/gef/updates/milestones
Adding repository https://download.eclipse.org/nebula/snapshot
Calculating requirements and dependencies.
Cannot complete the request.  Generating details.
ERROR: org.eclipse.pde.core code=0 Problems occurred while resolving the target contents
  at org.eclipse.oomph.util.OomphPlugin.coreException(OomphPlugin.java:291)
  at org.eclipse.oomph.util.pde.TargetPlatformUtil.activateTargetDefinition(TargetPlatformUtil.java:151)
  at org.eclipse.oomph.targlets.internal.core.TargletContainer.forceUpdate(TargletContainer.java:803)
  at org.eclipse.oomph.setup.targlets.impl.TargletTaskImpl$4.run(TargletTaskImpl.java:1174)
  at org.eclipse.oomph.util.pde.TargetPlatformUtil.runWithTargetPlatformService(TargetPlatformUtil.java:120)
  at org.eclipse.oomph.setup.targlets.impl.TargletTaskImpl.perform(TargletTaskImpl.java:1035)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.doPerformNeededSetupTasks(SetupTaskPerformer.java:3851)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.access$1(SetupTaskPerformer.java:3794)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer$WorkspaceUtil$1.run(SetupTaskPerformer.java:5178)
  at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2313)
  at org.eclipse.core.internal.resources.Workspace.run(Workspace.java:2338)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer$WorkspaceUtil.performNeededSetupTasks(SetupTaskPerformer.java:5172)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer$WorkspaceUtil.access$0(SetupTaskPerformer.java:5170)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.performNeededSetupTasks(SetupTaskPerformer.java:3785)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.performTriggeredSetupTasks(SetupTaskPerformer.java:3760)
  at org.eclipse.oomph.setup.internal.core.SetupTaskPerformer.perform(SetupTaskPerformer.java:3638)
  at org.eclipse.oomph.setup.ui.wizards.ProgressPage$9.run(ProgressPage.java:600)
  at org.eclipse.oomph.setup.ui.wizards.ProgressPage$11$1.run(ProgressPage.java:727)
  at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
  ERROR: org.eclipse.equinox.p2.director code=10053 Cannot complete the install because one or more required items could not be found.
    ERROR: org.eclipse.equinox.p2.director code=0 Software being installed: artificial_root 1.0.0.v1645017302048
    ERROR: org.eclipse.equinox.p2.director code=0 Missing requirement: artificial_root 1.0.0.v1645017302048 requires 'org.eclipse.equinox.p2.iu; org.eclipse.platform.releng.tychoeclipsebuilder.plain.project 0.0.0' but it could not be found

Took 7 seconds.
There are failed tasks.
Press Back to choose different settings or Cancel to abort.
`

**Comment by merks** [comment]
I have changed the setups, but of course you have cloned the old clones and those have been modified to remove all the contents of the repository.  So it's kind of tricky to fix the installation. 

Firstly, you need to do the Perform setup tasks, click Previous,  then select show all, and find the two clones that have moved to github and change the URI to the github choice.  You also need to delete the clones so that new ones are created.  It's quite tricky...

**Comment by jukzi** [comment]
Hi Ed, i loved OOmph because it was almost a single click solution. I never got into the details. How do i "perform setup tasks"? Why can't i just start somehow from scratch with a single click solution again?

**Comment by mickaelistria** [comment]
> Please, when you move repositories update all references to the new location.

One thing to keep in mind is that some of these references are not used as part of official builds nor actually required. So it's easy that people performing the migration can miss some reference for those references they are not even aware of.
I'm editing the instruction messages to add a note about migrating Oomph and .psf stuff. More concrete info about that would be welcome, and fixes to those files as well.

**Comment by jukzi** [comment]
@vogella you should update https://www.vogella.com/tutorials/EclipsePlatformDevelopment/article.html and https://www.slideshare.net/LarsVogel/eclipse-committerin20minutes too.
@mickaelistria the most important task would be to **test** all tutorials, installers and wikis if they still work after every move. it may help to google the old repo name to search for outdated references.
@merks how to delete all cached/save/remembered things to start from scratch as needed? And that step should make it into the wiki too.

**Comment by merks** [comment]
Note that using Navigate -> Open Setup -> User you'll see information like this recorded in the User setup:

![image](https://user-images.githubusercontent.com/208716/154317858-b9208de2-f948-4d67-95d4-ab97c951e109.png)

I've selected the particular nodes that record you choice of URI for the git clones that have moved.

Again, when you are on the Variables page using the "Show all variables" check box can you can change any of these recorded variable.  Note that the name of the variable is derived from the ID on the Git clone task which is I why I mention that changing the ID would help you get prompted for a new variable.

Unfortunately I'm traveling then next few weeks and it will be almost impossible to keep up with this rapidly migrating clone locations.

**Comment by laeubi** [comment]
> but of course you have cloned the old clones and those have been modified to remove all the contents of the repository.

I don't think that `remove all contents` is a good idea. Why are they not just simply made into mirrors of the github content for a while, or simply kept at last state with adding a clear note that they are relocated?

**Comment by merks** [comment]
Yes, it seems not a good idea. It corrupts existing environments and makes this migration process more painful than it needs to be...

> > but of course you have cloned the old clones and those have been modified to remove all the contents of the repository.
> 
> I don't think that `remove all contents` is a good idea. Why are they not just simply made into mirrors of the github content for a while, or simply kept at last state with adding a clear note that they are relocated?

**Comment by laeubi** [comment]
By the way (not kno if it is possible) what should work ist to edit the exiting reproistory and replace the old remote with the new remote URL.

**Comment by sravanlakkimsetti** [comment]
> By the way (not kno if it is possible) what should work ist to edit the exiting reproistory and replace the old remote with the new remote URL.

This is what I did to my existing clones and this works well

**Comment by laeubi** [comment]
> This is what I did to my existing clones and this works well

@merks could you add some magic to Ommpf (e.g. a "remote migration task" to accomplish something like this with a setup file?

**Comment by iloveeclipse** [comment]
Note, master branch on both repos is unfortunately diverged, so once pulled the latest state from old branch one cannot just switch remote "automatically"  get the latest from new repo. One has to hard reset to first common ancestor first.

To prevent this, the commit that adds the note about move must be done before forking, not after.

**Comment by laeubi** [comment]
> Starting from scratch is easier,

Is this still current? https://wiki.eclipse.org/Eclipse_Platform_SDK_Provisioning if I use the setup it still references the old git repositories.. DO I need to update/clear some cache?

**Comment by mickaelistria** [comment]
> @mickaelistria the most important task would be to test all tutorials, installers and wikis if they still work after every move. it may help to google the old repo name to search for outdated references.

FWIW, I don't believe anyone has the time to lookup all outdated tutorials on the web and even less time to try them, particularly if I don't need/use them. It's up to the tutorials maintains to take care of their stuff. We can for sure try to fix some Eclipse project doc by the way, any help is welcome towards that.

> Why are they not just simply made into mirrors of the github content for a while, or simply kept at last state with adding a clear note that they are relocated?

The "simply" here seems a bit too naive and optimistic. Setting up a Git close is not trivial and is not perfect for users as well. Projects that manage clones (Eclipse Platform had done it for some time a few years ago) face new problems, with synchronization or unclear contribution path.

> By the way (not kno if it is possible) what should work ist to edit the exiting reproistory and replace the old remote with the new remote URL.

We can change the MOVED_TO_GITHUB file to include such instructions.

> I don't think that remove all contents is a good idea

Some other alternative is to ask EF to simply block the GitHub repo, but then people getting this repo don't get any idea it is moved and basically stick to older repo by absence of hint/annoyance to make them learn about the change.
Another one is to ask EF to remove the repo, but all previous links will be dead. I think this one may be better than adding an extra MOVED_TO_GITHUB commit.

> Note, master branch on both repos is unfortunately diverged

It's not unfortunate. If we want people to react to the migration, they first need to know about the migration, and the divergence is the alert.

**Comment by akurtakov** [comment]
> > I don't think that remove all contents is a good idea
> 
> Some other alternative is to ask EF to simply block the GitHub repo, but then people getting this repo don't get any idea it is moved and basically stick to older repo by absence of hint/annoyance to make them learn about the change. Another one is to ask EF to remove the repo, but all previous links will be dead. I think this one may be better than adding an extra MOVED_TO_GITHUB commit.

Note that old repos will have to stay for old branches as no one will spend the time to fix building them with new changes!

**Comment by iloveeclipse** [comment]
> Note that old repos will have to stay for old branches as no one will spend the time to fix building them with new changes!

Alex, can you explain that? I haven't understand why keeping the code in old repo breaks anything.
Removing the code makes "automatic" remote switch impossible and breaks users that aren't git experts.

> It's not unfortunate. If we want people to react to the migration, they first need to know about the migration, 

I would expect announcing migration on mailing lists should be enough. People not on a mailing list will realize looking on the project page that would say that "our repo is moved, please update your remote".

> and the divergence is the alert.

This is not an alert, but a pain and hinders people in the migration because first they pull broken state, are now "alerted" and after that they have to switch remote and **manually** reset branch back to some "right" commit - which is error prone. or they will try to merge and will have some crazy state after that.

**Comment by mickaelistria** [comment]
> People not on a mailing list will realize looking on the project page that would say that "our repo is moved, please update your remote".

Which project page? Do you really believe that people who have cloned a repo once and successfully worked with it for years will suddenly start monitoring some project page weekly "just in case there is something new to know about"? That seems way too optimistic to me.

**Comment by akurtakov** [comment]
> > Note that old repos will have to stay for old branches as no one will spend the time to fix building them with new changes!
> 
> Alex, can you explain that? I haven't understand why keeping the code in old repo breaks anything. Removing the code makes "automatic" remote switch impossible and breaks users that aren't git experts.

Keeping the code is not breaking anything, what would break it is making the repo READ_ONLY so people can't submit gerrit against it.
All the maintenance streams (https://git.eclipse.org/c/platform/eclipse.platform.releng.aggregator.git/refs/heads) will most probably stay on gerrit unless some brave heart decides to go and fix all configs in the streams whenever a repo moves which would a be a whole lot of work . 
It's another discussion (which I definetely don't want to start!) whether all these maintenance streams should be preserved at all. Keeping things for people in deep maintenance on old infra sounds best and once old infra is seized it's up to them to spend the time to migrate - whenever and if they feel about it.

**Comment by iloveeclipse** [comment]
> Which project page? Do you really believe that people who have cloned a repo once and successfully worked with it for years will suddenly start monitoring some project page weekly "just in case there is something new to know about"?

No, I assume people who cloned original repo once and wonder why they don''t see latest state would google for "eclipse project", click on developer resources and see that the repo is moved on  
https://projects.eclipse.org/projects/eclipse.platform/developer.

Then simply changing remote & pull would work. 
If there would be a command example `git remote add github <url> ; git pull github master` on that page, it would be even better.

> Regarding not git expert users - I fix my clones by rm -fr repo; git clone github_repo - not git expertise needed

Sure, but not all people are familiar with all that stuff, some just use Oomph and are happy. 
So Oomph can most likely add or change a remote automatically, but Oomph **for sure** can't `git reset --hard` or `rm -rf` anything!

**Comment by mickaelistria** [comment]
> No, I assume people who cloned original repo once and wonder why they don''t see latest state

How do they get a hint that this is not the latest state? Their last fetch was as successful as all other ones they've done so far, so their is nothing hinting them to wonder about the state.

> Sure, but not all people are familiar with all that stuff, some just use Oomph and are happy.

Developers need to be familiar with industry-standard Git. It's a must-have nowadays. We cannot put wrapper on top of industry-standards expecting it prevents people from being fluent with industry standard. Anyone who's not able to add a new remote in git, to do a reset or to cherry-pick a patch needs to seriously consider reading Git tutorals ASAP. It's not Eclipse project's fault if some people didn't learn Git properly, just as much as it's not Eclipse's fault if some people don't know how to code in Java, it's a pre-requisite and that's all.

**Comment by jukzi** [comment]
It's not the users fault if he tries something that was valid some time ago. He needs a information what he should do or at least what happened. How about renaming the old repos or their urls somehow (for example postfix "_backup") so that is obvious to the user that the remote repository does not exist anymore (but still we have a backup). 
Well and i am happy that i rarely have to configure something on git and i am sure most of my colleagues don't know how to configure git because it's done for them automatically.

**Comment by mickaelistria** [comment]
> How about renaming the old repos or their urls somehow (for example postfix "_backup") so that is obvious to the user that the remote repository does not exist anymore (but still we have a backup).

See https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/issues/24#issuecomment-1042803096

**Comment by iloveeclipse** [comment]
> How do they get a hint that this is not the latest state?
Easy. They could see that there are no commits for a months and the last commit has commit message "THIS REPO IS MOVED TO GITHUB".

> Developers need to be familiar with industry-standard Git. It's a must-have nowadays. We cannot put wrapper on top of industry-standards expecting it prevents people from being fluent with industry standard. 

I believe that could be also the answer to "Let move platform to C++/assembler, that is industry standard for years and why do we need a Java wrapper for the standard?".

I'm developing a lot in Eclipse, but I'm neither git or maven or p2 or e4 guru, even if the technologies are "industry standards" for years. This doesn't prevent me contributing, and expecting that I have to learn that all is not realistic, I have other things to do in my day job.

So if you are saying that "move to github will make contribution much easier" you should make the transition **easy too**, otherwise you will loose more then gain.

In our case it would be: 
1) Non-destructive commit on original repo clearly saying "Development is moved to http://github.com/path/to/new/repo".
2) Fork of original repo to github
3) Message on the mailing lists announcing the move and providing git command to perform remote switch.
4) Note on the project page that repository is moved, again with the transition hint / command.
5) **Optionally** Some clever oomph setup task that can switch old repo remote to the new one or simply add a new one.

Looking on the steps above - that isn't a big deal to do (except 5), compared with what was already done for moved repos? It is just few minutes to write a message templates for everything and the only effort will be to copy/paste & enter right url into the git commits / mails / wikis? And we can hope that @merks may be have some genius idea how to implement 5) with Oomph for users. 

But of course simply deleting the old data and switch development to the github repo is easier.

**Comment by mickaelistria** [comment]
> Non-destructive commit on original repo clearly saying "Development is moved to [http://github.com/path/to/new/repo](https://github.com/path/to/new/repo)".

Again, I'm still afraid no-one will notice it and people will still push to Gerrit. The reality is that people don't change unless you force them to, the destructive commit is a way to force them: they'll get some conflicts, see the MOVED_TO_GITHUB file, read it and react.

> Message on the mailing lists announcing the move and providing git command to perform remote switch.

I edited the migration steps to include such a notification.

> Note on the project page that repository is moved, again with the transition hint / command.

We cannot easily add notes there, but the migration steps include updating the project page to remove Gerrit repo and show GitHub one.

**Comment by iloveeclipse** [comment]
> no-one will notice it and people will still push to Gerrit.
I'm pretty sure one argument for moving to github was that "no one understands how to use gerrit". 

How likely is that people that did all the setup **manually**
1) People don't notice a commit on master "Development is moved to ..." before committing
2) Don't get any reply on bugzilla / gerrit change
3) Don't check mails / wiki's
4) Are unable to use google and check the project state

Very unlikely.
However 

How likely is that people that did all the setup **via Oomph** and have no ideas about git/repo cloning etc will
1) Pull latest broken (diverged) repo state
2) Will not be able to solve the problem via Oomph / automatically
3) Will have to go down to command line / try to reverse engineer what the Oomph did and where they have to run which commands?

What is the actual problem we are trying to solve with "breaking" the old repository state? Only to make sure people get notified? Isn't that too big hammer? What "bad" would happen if we don't do so? Nothing.

So we have:
1) approach one: break repo and everyone need to manually solve the problems by yourself with unclear effort / risk to damage existing data.
2) approach two: don't break the old repo, and people can switch to the new one in few clicks / key strokes without the risk to damage their data.

Effort for **us** is almost same for both. Effort for users is clearly smaller for the second approach.

**Comment by mickaelistria** [comment]
I'm removing the step about modifying the Gerrit repo alltogether to not disturb the "do not disturb" camp. But beware that it will make that several people will still push to Gerrit after migration because they won't be aware of the change and nothing will have automatically interrupted them in doing so. I hope you'll have the same energy along the multiple disturbances those users will cause to provide them assistance than you're showing here to argue we shouldn't disturb right now. In the end, there will be disturbance, the question is how we distribute the disturbance over people and time. A break is disturbing users once per repo, a continuation is disturbing committers as long as not all contributors have understood a migration has happened.

**Comment by merks** [comment]
> > This is what I did to my existing clones and this works well
> 
> @merks could you add some magic to Ommpf (e.g. a "remote migration task" to accomplish something like this with a setup file?

Not without significant new effort.  The git clone task is not designed to modify clone once it exists.    One can modify the URI used by the clone task so that it uses the right one in the future and can modify the clone itself...

**Comment by akurtakov** [comment]
Eclipse PDE organization requested at https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/1006

**Comment by merks** [comment]
FYI, I've been updating the various setups as the repos are migrated.  To avoid the problems described above, I've also been changing the ID of each git clone task so that the user will be prompted for the github URL to be used for the clone, rather than the old git.eclipse.org URI being reused and needing to be changed manually.  This seems to be working well...

**Comment by laeubi** [comment]
> The transition process is painful, as we've all noticed...

I don't think so, just wanted to make sure that I understand it correctly.

> Note that I do test that a newly created setup works properly.

Thanks for all the work :+1:

**Comment by merks** [comment]
I noticed that *org.eclipse.platform.news* is not in the list of repositories to migrate above.  I assume that's an oversight.

This is the current result of the setups:

![image](https://user-images.githubusercontent.com/208716/160595184-d490addd-2c8d-40b3-b495-2eac52154f82.png)

**Comment by laeubi** [comment]
> you need to do manually whatever you need to do manually without Oomph.

I just tried it out with the p2 repo and it is not that many manual work I do the following:

1. remove the old repository
2. select "remove projects connected to this repository"
3. the I install al updates and perform setup tasks
4. Now everything is migrated

**Comment by merks** [comment]
Yes, that works too, though often on Windows the repository folder is locked (or rather some file in it is locked/open) and then the entire folder cannot be removed without a restart...

**Comment by iloveeclipse** [comment]
Hi,
I'm unable to see "Fetch Github Pull request" for any of my github configured platform repository. 

I'm trying to get https://github.com/eclipse-platform/eclipse.platform.resources/pull/15 into local workspace **without** manually adding remote repo and **manually** configuring fetch, so that I can test the patch locally. This was a no-brainer task with gerrit.

I read 
- https://bugs.eclipse.org/bugs/show_bug.cgi?id=544815
- https://wiki.eclipse.org/EGit/New_and_Noteworthy/6.0#Fetching_Pull_Requests 
and using latest nightly egit version.

What I'm missing? I can right click on any node in the egit git repo view (repo itself, my fork remote, fetch, push nodes), there is no context menu offering "pull from github". Do I need to configure server url, and how it should look like?
If I use Ctrl+3 to open the "Fetch Github Pull Request" wizard, it opens and shows "refs/pull/15/head" in the "Pull request"  field but nothing in the  "URI" field. Only if I **manually** enter https://github.com/eclipse-platform/eclipse.platform.resources into initially **empty** "URI" field, it works.

Push/Pull works fine, and I've already pushed/pulled few commits to the same repo via EGit.
So obviously something is either wrong configured or I'm missing something trivial to get "Fetch PR" menu / Wizard working.

**Comment by laeubi** [comment]
I use the official SDK setup and get the following in the git view:
![grafik](https://user-images.githubusercontent.com/1331477/160601243-c7514cdc-9d4c-43cf-b79a-1145b37c9f88.png)

**Comment by merks** [comment]
Yes, when I have `https://github.com/eclipse-equinox/p2/pull/21` in the clipboard and open that dialog for p2

![image](https://user-images.githubusercontent.com/208716/160605096-10e75a21-2980-42af-9fde-5b512aba1fe2.png)

the dialog is filled like this:

![image](https://user-images.githubusercontent.com/208716/160604737-aa8f74d0-97c7-4080-89e6-b2606d941957.png)

**Comment by iloveeclipse** [comment]
OK, I've compared the urls used in pom.xml with my workspace project, and I see that I didn't had `.git` at the end of my remotes (don't ask me why).
```
git remote -v
iloveeclipse    https://github.com/iloveeclipse/eclipse.platform.resources (fetch)
iloveeclipse    https://github.com/iloveeclipse/eclipse.platform.resources (push)
origin  https://github.com/eclipse-platform/eclipse.platform.resources (fetch)
origin  https://github.com/eclipse-platform/eclipse.platform.resources (push)
```
After changing to the url ending with `.git` the context menu shows up **on the repo node only**.
So even if push/pull/fetch works without issues, the context menu seem to require `.git` in the server url.
This below works:
```
git remote set-url origin https://github.com/eclipse-platform/eclipse.platform.resources.git
git remote set-url --push origin https://github.com/eclipse-platform/eclipse.platform.resources.git
git remote -v
iloveeclipse    https://github.com/iloveeclipse/eclipse.platform.resources (fetch)
iloveeclipse    https://github.com/iloveeclipse/eclipse.platform.resources (push)
origin  https://github.com/eclipse-platform/eclipse.platform.resources.git (fetch)
origin  https://github.com/eclipse-platform/eclipse.platform.resources.git (push)
```

**Comment by laeubi** [comment]
> the context menu shows up **on the repo node only**

I noticed this as well, you could ask [Egit](https://bugs.eclipse.org/bugs/enter_bug.cgi?product=EGit) to also show it on the project level, and even if the url starts with https:/github.com/... 

I think this is all about everyone needs to give feedback on how we can improve the workflow, also see [Bug 579458](https://bugs.eclipse.org/bugs/show_bug.cgi?id=579458) for example

**Comment by iloveeclipse** [comment]
Created bug https://bugs.eclipse.org/bugs/show_bug.cgi?id=579477

**Comment by sravanlakkimsetti** [comment]
> I noticed that _org.eclipse.platform.news_ is not in the list of repositories to migrate above. I assume that's an oversight.
> 
> 

I believe you referring to 
https://git.eclipse.org/c/www.eclipse.org/eclipse/news.git/
https://git.eclipse.org/c/www.eclipse.org/eclipse.git/

**Comment by merks** [comment]
Yes, those two need attention too.

The following exist as well and I assume should also be migrated

https://git.eclipse.org/c/www.eclipse.org/jdt.git/  -> https://www.eclipse.org/jdt/
https://git.eclipse.org/c/www.eclipse.org/pde.git/ -> https://www.eclipse.org/pde/
https://git.eclipse.org/c/www.eclipse.org/equinox.git/ -> https://www.eclipse.org/equinox/

It looks like https://www.eclipse.org/rt/ disappeared although https://git.eclipse.org/c/www.eclipse.org/rt.git/ still exists.

**Comment by iloveeclipse** [comment]
Note for the upcoming repositories to be converted: whoever requests that, please remind the admins to protect branches master and R*maintenance to  "Require a pull request before merging".
See https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/1088

**Comment by sravanlakkimsetti** [comment]
> Yes, those two need attention too.
> 
> The following exist as well and I assume should also be migrated
> 
> https://git.eclipse.org/c/www.eclipse.org/jdt.git/ -> https://www.eclipse.org/jdt/ https://git.eclipse.org/c/www.eclipse.org/pde.git/ -> https://www.eclipse.org/pde/ https://git.eclipse.org/c/www.eclipse.org/equinox.git/ -> https://www.eclipse.org/equinox/
> 
> It looks like https://www.eclipse.org/rt/ disappeared although https://git.eclipse.org/c/www.eclipse.org/rt.git/ still exists.

Publishing of website from github repo is not yet available. We created couple of jenkins jobs to copy contents from github to gerrit repo. Lets wait till foundation comes up with a publishing solution before migration others. Opened #182 to track migration of website repositories.

**Comment by mickaelistria** [comment]
I think we can now consider this one as resolved.
Some more question have and will be raised, but it's not better to track them in dedicated discussions.

**Comment by HannesWell** [comment]
Thank you all that were involved for this smooth migration. 🎉👍🏽

**Comment by merks** [comment]
The migration of the all website repositories is not complete.  I had added the following at the top, but a subsequent edit removed them again:

https://github.com/eclipse-equinox/.github/issues/1
https://github.com/eclipse-jdt/.github/issues/1
https://github.com/eclipse-pde/.github/issues/1

It seems editing comments is a bit error prone.  Perhaps one must be very careful to refresh the page to make sure you edit the latest version....

I don't know the status of the platform project's website migration?  I know they are migrated, but is the process of publishing them to the Eclipse website working now?  Somehow the links for those disappeared from the top as well...

Could someone please provide update?

**Comment by sravanlakkimsetti** [comment]
> The migration of the all website repositories is not complete. I had added the following at the top, but a subsequent edit removed them again:
> 
> [eclipse-equinox/.github#1](https://github.com/eclipse-equinox/.github/issues/1) [eclipse-jdt/.github#1](https://github.com/eclipse-jdt/.github/issues/1) [eclipse-pde/.github#1](https://github.com/eclipse-pde/.github/issues/1)
> 
> It seems editing comments is a bit error prone. Perhaps one must be very careful to refresh the page to make sure you edit the latest version....
> 
> I don't know the status of the platform project's website migration? I know they are migrated, but is the process of publishing them to the Eclipse website working now? Somehow the links for those disappeared from the top as well...
> 
> Could someone please provide update?

We have 5 website repositories
2 from platform and 3 for other 3 orgs. 

The publishing of websites is from gerrit repos only. We don't have way to publish websites using github repos yet. 
We did migrate platform's website repositories, at the same time I had to create couple of jobs to synchronize(copy commits from) github repo to gerrit repo. Due to this I had to postpone migration of website repositories till that time we have a solution for publishing from github repos. 
You can track this effort in #182 

The sync jobs are working now. So publishing works fine for the platform repositories.
